File: 181 excersises/173.cpp
#include <iostream>
using namespace std;

bool esNumeroPerfecto(int numero) {
    if (numero <= 1) {
        return false;
    }

    int sumaDivisores = 1;
    
    for (int i = 2; i <= numero / 2; i++) {
        if (numero % i == 0) {
            sumaDivisores += i;
        }
    }
    
    return sumaDivisores == numero;
}

int main() {
    int numero;
    cout << "Ingresa un numero para revisar si es perfecto: ";
    cin >> numero;
    
    if (esNumeroPerfecto(numero)) {
        cout << numero << " es un numero perfecto ✨." << endl;
    } else {
        cout << numero << " no es un numero perfecto." << endl;
    }
    
    return 0;
}

File: 181 excersises/173.py
def es_numero_perfecto(numero):
    if numero <= 1:
        return False
    
    suma_divisores = 1
    
    for i in range(2, numero // 2 + 1):
        if numero % i == 0:
            suma_divisores += i
    
    return suma_divisores == numero

numero = int(input("Ingresa un numero para checkear si es perfecto: "))

if es_numero_perfecto(numero):
    print(f"{numero} es un número perfecto ✨.")
else:
    print(f"{numero} no es un número perfecto.")